lib/tasks/data/truncate_data.rake

```diff
@@ -1,81 +1,28 @@
-
-      sql_delete_from "spree_line_items #{where_order_id_in_orders_to_delete}"
-      sql_delete_from "spree_payments #{where_order_id_in_orders_to_delete}"
-      sql_delete_from "spree_shipments #{where_order_id_in_orders_to_delete}"
-      Spree::ReturnAuthorization.delete_all
-
-      truncate_order_cycle_data
-
-      sql_delete_from "proxy_orders #{where_oc_id_in_ocs_to_delete}"
-
-      sql_delete_from "spree_orders #{where_oc_id_in_ocs_to_delete}"
-      sql_delete_from "order_cycle_schedules #{where_oc_id_in_ocs_to_delete}"
-      sql_delete_from "order_cycles #{where_ocs_to_delete}"
-
-      Spree::TokenizedPermission.where("created_at < '#{date}'").delete_all
-      Spree::StateChange.delete_all
-      Spree::LogEntry.delete_all
-      sql_delete_from "sessions"
-    end
-
-    def sql_delete_from(sql)
-      ActiveRecord::Base.connection.execute("delete from #{sql}")
-    end
-
-    private
-
-    def date
-      3.months.ago
-    end
-
-    def where_ocs_to_delete
-      "where orders_close_at < '#{date}'"
-    end
-
-    def where_oc_id_in_ocs_to_delete
-      "where order_cycle_id in (select id from order_cycles #{where_ocs_to_delete} )"
-    end
-
-    def where_order_id_in_orders_to_delete
-      "where order_id in (select id from spree_orders #{where_oc_id_in_ocs_to_delete})"
-    end
+    task :truncate, [:months_to_keep] => :environment do |_task, args|
+      warn_with_confirmation
 
-    def truncate_adjustments
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Order'
-        and source_id in (select id from spree_orders #{where_oc_id_in_ocs_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Shipment'
-        and source_id in (select id from spree_shipments #{where_order_id_in_orders_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Payment'
-        and source_id in (select id from spree_payments #{where_order_id_in_orders_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::LineItem'
-        and source_id in (select id from spree_line_items #{where_order_id_in_orders_to_delete})"
+      months_to_keep = args.months_to_keep.to_i
+      TruncateData.new(months_to_keep: months_to_keep).call
     end
 
-    def truncate_order_cycle_data
-      sql_delete_from "coordinator_fees #{where_oc_id_in_ocs_to_delete}"
-      sql_delete_from "
-        exchange_variants where exchange_id
-        in (select id from exchanges #{where_oc_id_in_ocs_to_delete})"
-      sql_delete_from "
-        exchange_fees where exchange_id
-        in (select id from exchanges #{where_oc_id_in_ocs_to_delete})"
-      sql_delete_from "exchanges #{where_oc_id_in_ocs_to_delete}"
+    def warn_with_confirmation
+      message = <<-MSG.strip_heredoc
+      \n
+      <%= color('This will permanently change DB contents. Please, make a backup first.', :yellow) %>
+      Are you sure you want to proceed? (y/N)
+      MSG
```

Comment:
This now looks like

![Screenshot from 2020-03-06 17-56-57](https://user-images.githubusercontent.com/762088/76104566-e56bb200-5fd3-11ea-9bbc-c2ae43e8c873.png)

Reply:
Is there a color like :blinking_and_large_red instead of :yellow :-)

Reply:
bold red will do the trick :trollface:

Reply:
LOL https://github.com/JEG2/highline/blob/bf489b632afd92c68df0b2c1f348afb66b5495aa/test/test_highline.rb#L475

Reply:
ahah, now we have to use it :-)